test/vendor/utils.test.js

```diff
@@ -146,4 +157,88 @@ describe('utils', () => {
     expect(missing(0)).toBe(false);
     expect(missing({})).toBe(false);
   });
+
+  it('first', () => {
+    expect(first('test')).toBe('t');
+    expect(first('jest')).toBe('j');
+    expect(first(['11', '22', '33'])).toBe('11');
+    expect(first('')).toBe(null);
+  });
```

Comment:
Sorry for some delay. I'm tried to understand the best way how to fix it, but I'm not sure. Can you share your vision about this cases:
1. last('') should be null or ''?
2. do we need handle type of the params to be sure that they are string or array? For example throw the error?

Reply:
Also I'm thinking why don't you use Lodash here to do this operations? Is that because Lodash will _.head('test') === 't'? If you want, I can try to refactor all this utils.

Reply:
@wicked-kiD 

1. `last('')===null`  since the empty string is treated as `missing`
2. You may need to check if the parameter types to return the proper value; or you can leverage other functions to do type-check for you:  (inefficient, but works)

```
    last = x => {
        const list = toArray(x);
        return list[list.length-1]
    }
```

In general, this library is simplifying data manipulation by establishing a set of "equivalence classes" on data: Different data structures that represent the same thing should be equivalent.

Javascript has no operator for equivalent, so I will use `<=>`

The `missing()` function defines all the values that are equivalent to `null`:

    null <=> '' <=> [] <=> Number.POSITIVE_INFINITY  ...

This library also assumes everything is a list:

    null <=> []
    x <=> [x]
    "hello" <=> ["hello"]

The expectation is this library will provide the left-most as the canonical representative of the class.

> why don't you use Lodash here to do this operations


This library started because the previous owner did not like the size of lodash.   You are welcome to replace part of this library with lodash and see what happens to the compiled size and how it impacts the other code.  You are right there may be opportunities there.